### ISSUE TITLE
Training speed on multi gpu

### ISSUE THREAD

**Issue by khanhnt** [clovaai/deep-text-recognition-benchmark#151]
Hi @ku21fan and all,

I have tried to fine-tune the model on 3 GPUs, but it seem the training speed is slow. With 16 epochs, the Elapsed_time is 35945.77589, that mean each epochs cost about 2250 seconds. Is it normal ? Besides, In order to train the model, I have to reduce the batch-size to 60. 
Here is the training options:
```
{experiment_name}: {TPS-ResNet-BiLSTM-Attn-Seed1111}
{train_data}: {result/train_06032020/train/}
{valid_data}: {result/train_06032020/valid/}
{manualSeed}: {1111}
{workers}: {12}
{batch_size}: {180}
{num_iter}: {300000}
{valInterval}: {2000}
{saved_model}: {model/TPS-ResNet-BiLSTM-Attn-case-sensitive_0.pth}
{FT}: {True}
{adam}: {False}
{lr}: {1}
{beta1}: {0.9}
{rho}: {0.95}
{eps}: {1e-08}
{grad_clip}: {5}
{select_data}: {['/']}
{batch_ratio}: {['1']}
{total_data_usage_ratio}: {1.0}
{batch_max_length}: {25}
{imgH}: {32}
{imgW}: {100}
{rgb}: {False}
{character}: {0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ!"#$%&'()*+,-./:;<=>?@[\]^_`{|}~áàảãạăắằẳẵặâấầẩẫậđéèẻẽẹêếềểễệíìỉĩịóòỏõọôốồổỗộơớờởỡợúùủũụưứừửữựỷỹỳýỵÁÀẢÃẠĂẮẰẲẴẶÂẤẦẨẪẬĐÉÈẺẼẸÊẾỀỂỄỆÍÌỈĨỊÓÒỎÕỌÔỐỒỔỖỘƠỚỜỐỖỘÚÙỦŨỤƯỨỪỬỮỰÝỲỶỸỴ}
{sensitive}: {True}
{PAD}: {False}
{data_filtering_off}: {False}
{Transformation}: {TPS}
{FeatureExtraction}: {ResNet}
{SequenceModeling}: {BiLSTM}
{Prediction}: {Attn}
{num_fiducial}: {20}
{input_channel}: {1}
{output_channel}: {512}
{hidden_size}: {256}
{num_gpu}: {3}
{num_class}: {230}
```
The training run on Intel(R) Xeon(R) CPU E5-2620 v3 @ 2.40GHz with 3 Tesla K80 :
```
| NVIDIA-SMI 440.33.01    Driver Version: 440.33.01    CUDA Version: 10.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  Tesla K80           Off  | 00000000:85:00.0 Off |                    0 |
| N/A   73C    P0   113W / 149W |   2783MiB / 11441MiB |     94%      Default |
+-------------------------------+----------------------+----------------------+
|   1  Tesla K80           Off  | 00000000:86:00.0 Off |                    0 |
| N/A   53C    P0   121W / 149W |   2155MiB / 11441MiB |     88%      Default |
+-------------------------------+----------------------+----------------------+
|   2  Tesla K80           Off  | 00000000:89:00.0 Off |                    0 |
| N/A   62C    P0   113W / 149W |   2145MiB / 11441MiB |     86%      Default |
+-------------------------------+----------------------+----------------------+

```

**Comment by khoon485** [comment]
how about more batch size? I guess you can allocate more batches. It might reduce total learning time.

**Comment by khanhnt** [comment]
Thanks @khoon485, I have tried to increase batch size, but it stuck, so I have to reduce it.

**Comment by khoon485** [comment]
Please don't forget you also have to reduce iteration number when you allocate more batches

**Comment by khanhnt** [comment]
Is it fine ? because according to this one [https://github.com/clovaai/deep-text-recognition-benchmark/blob/8e7aba3b5da27690b4216d16b87c2d5928f18709/train.py#L307](previous version), the author have commeted it. It was belong to the old version. 
Anyway I will try it, because after each epoch, the training will save the best accuracy one.  Thanks.

**Comment by freedom9393** [comment]
@khanhnt , could you please tell me how did you train with 3 GPUs?

**Comment by khanhnt** [comment]
@freedom9393 you need to specify the GPU id while launching the trainign command. For example, you have three GPUs, with the ID is 0,1,2. So the command is CUDA_VISIBLE_DEVICES=0,1,2 python3 train.py ...

**Comment by freedom9393** [comment]
@khanhnt , I run that. But it freezes, keep waiting like forever. I set batch size to 2, but still same, keep waiting..

**Comment by devarshi16** [comment]
Apologies for addressing super old issue. In case it's still relevant to anyone. I think I know what might help.

Try setting the argument `--workers 0` for multi-gpu settings. In case the validation intervals are taking too long to print you can manually reduce them from the default value of `2000` intervals using the argument `--valInterval 500` you can also reduce your total iterations if you have a high batch size using the argument `--num_iter`.
